.github/workflows/swift_build.yml

```diff
@@ -0,0 +1,22 @@
+name: Swift CI on PR
+
+on:
+  pull_request:
```

Comment:
should this be push?

Reply:
I guess we should have push as well as pull request. 
Push will trigger when you push code where pull_request will trigger when there is a pull request.
We probably want to do both to main. Do we agree?